lib/src/listing/listing_generator.dart

```diff
@@ -34,30 +34,34 @@ abstract class ListingGenerator {
     final kLimitKey = 'limit';
     final kAfterKey = 'after';
     final nullLimit = 1024;
-    final paramsInternal = params ?? new Map();
+    final paramsInternal = (params != null) ? new Map.from(params) : new Map();
```

Comment:
logic is a bit hard to read, should we put:
`final paramsInternal = (params == null) ? new Map() : new Map.from(params);`

Reply:
Done.